lib/analysis/modules/ies_enkf.c

```diff
@@ -278,26 +275,25 @@ void ies_enkf_updateA( void * module_data,
 *     Copies the initial measurement perturbations for the active observations into the current E matrix.
 *     Copies the inputs in D, Y and R into their local representations
 */
-   matrix_type * Y   = matrix_alloc( nrobs    , ens_size );
-   matrix_type * E   = matrix_alloc( nrobs    , ens_size );
-   matrix_type * D   = matrix_alloc( nrobs    , ens_size );
-   matrix_type * R   = matrix_alloc( nrobs    , nrobs );
-   matrix_type * D0  = matrix_alloc_copy( Din );
+   matrix_type * Y   = matrix_alloc_copy( Yin );
+   matrix_type * E   = matrix_alloc( nrobs_inp, ens_size );
+   matrix_type * D   = matrix_alloc_copy( Din );
+   matrix_type * R   = matrix_alloc_copy( Rin );
 
 /* Subtract new measurement perturbations              D=D-E    */
-   matrix_inplace_sub(D0,Ein);
+   matrix_inplace_sub(D,Ein);
 
 /* Extract active observations and realizations for D, E, Y, and R    */
```

Comment:
The comment needs an update here

Reply:
The comment is correct. I just removed D0 since it, in the new code is identical to D.